File: go/problems/day17_test.go
package problems

import "testing"

var day17 Solvable[int] = Day17{}

func TestDay17_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day17_1")
	result := day17.Solve(input)

	var expected int = 102
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay17_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day17_2")
	result := day17.Solve(input)

	var expected int = 970
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day17.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day17.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 17", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day17_1");
    const result = solve(input);
    assert.equal(result, 102);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day17_2");
    const result = solve(input);
    assert.equal(result, 970);
  });
});
